listenbrainz/webserver/static/js/src/utils.tsx

```diff
@@ -415,6 +421,53 @@ const pinnedRecordingToListen = (pinnedRecording: PinnedRecording): Listen => {
   };
 };
 
+const getAlbumArtFromListenMetadata = async (
+  listen: BaseListenFormat,
+  spotifyUser?: SpotifyUser
+): Promise<string | undefined> => {
+  // if spotifyListen
+  if (
+    SpotifyPlayer.isListenFromThisService(listen) &&
+    SpotifyPlayer.hasPermissions(spotifyUser)
+  ) {
+    const trackID = SpotifyPlayer.getSpotifyTrackIDFromListen(listen);
+    return new SpotifyAPIService(spotifyUser).getAlbumArtFromSpotifyTrackID(
+      trackID
+    );
+  }
+  if (YoutubePlayer.isListenFromThisService(listen)) {
+    const videoId = YoutubePlayer.getVideoIDFromListen(listen);
+    const images = YoutubePlayer.getThumbnailsFromVideoid(videoId);
+    return images?.[0].src;
+  }
+  /** Could not load image from music service, fetching from CoverArtArchive if MBID is available */
+  const releaseMBID = getReleaseMBID(listen);
+  if (releaseMBID) {
+    try {
+      const CAAResponse = await fetch(
+        `https://coverartarchive.org/release/${releaseMBID}`
+      );
+      if (CAAResponse.ok) {
+        const body: CoverArtArchiveResponse = await CAAResponse.json();
+        if (!body.images?.[0]?.thumbnails) {
+          return undefined;
+        }
+        const { thumbnails } = body.images[0];
+        return (
+          thumbnails[250] ??
+          thumbnails.small ??
+          // If neither of the above exists, return the fisrt one we find
+          // @ts-ignore
+          thumbnails[Object.keys(thumbnails)?.[0]]
+        );
+      }
+    } catch (error) {
+      // Do nothing with the error, I guess.
```

Comment:
I think we can log these errors here?  Or keep a record of these somewhere maybe?